lib/backends/sdc/affinity.js

```diff
@@ -0,0 +1,736 @@
+    } else if (aff.key !== 'container' && aff.valueType === 're') {
+        query = {
+            fields: 'uuid,alias,tags',
+            owner_uuid: opts.ownerUuid,
+            state: 'active',
+            predicate: JSON.stringify({
+                or: [
+                    {eq: ['tag.' + aff.key,              '*']},
+                    {eq: ['tag.docker:label:' + aff.key, '*']}
+                ]
+            })
+        };
+        opts.vmapi.listVms(query, {
+            headers: {'x-request-id': log.fields.req_id}
+        }, function (err, allVms) {
+            if (err) {
+                cb(err);
+                return;
+            }
+            vms = [];
+            for (i = 0; i < allVms.length; i++) {
+                vm = allVms[i];
+
+                var tag = vm.tags[aff.key];
+                if (tag !== undefined && aff.valueRe.test(tag.toString())) {
+                    // Docker labels can only be strings. Triton VM tags can
+                    // also be booleans or numbers.
+                    vms.push(vm);
+                    continue;
+                }
+                var label = vm.tags['docker:label:' + aff.key];
+                if (label !== undefined && aff.valueRe.test(label)) {
+                    vms.push(vm);
+                    continue;
+                }
+            }
+            log.debug({affinity: _strFromFilterExpr(aff), vms: vms},
+                '_vmUuidsFromAffinity');
+            var vmUuids = vms.map(function (vm_) { return vm_.uuid; });
+            cb(null, vmUuids);
+        });
+
+    // container==UUID
+    } else if (common.isUUID(aff.value)) {
+        assert.equal(aff.key, 'container');
+        cb(null, [aff.value]);
+
+    // container==<full 64-char docker id>
+    //
+    // Given a full 64-char docker id, Docker-docker will skip container
+    // *name* matching (at least that's what containers.js#findContainerIdMatch
+    // implies). We'll do the same here. Any other length means we need to
+    // consider name matching.
+    } else if (/^[a-f0-9]{64}$/.test(aff.value)) {
+        assert.equal(aff.key, 'container');
+        var vmUuid = common.dockerIdToUuid(aff.value);
+        opts.vmapi.getVm({
+            uuid: vmUuid,
+            owner_uuid: opts.ownerUuid,
+            fields: 'uuid,alias,state,internal_metadata,docker'
+        }, {
+            headers: {'x-request-id': log.fields.req_id}
+        }, function (err, vm_) {
+            if (err && err.statusCode !== 404) {
+                cb(err);
+            } else if (!err && vm_ && vm_.docker
+                && ['destroyed', 'failed'].indexOf(vm_.state) === -1
+                && vm_.internal_metadata['docker:id'] === aff.value)
+            {
+                /*
+                 * If the 'docker:id' check fails, treat this like a 404.
+                 * This could happen if someone made up a docker_id that
+                 * started with the 32-chars for a real VM.
```

Comment:
This block comment doesn't look like it's describing the code here. Does that belong in the else?